### PR TITLE
fix: fall back to old behavior if newer value is nil in core data tracker

### PR DESCRIPTION
**Pull request by armcknight**
Fixes a possible crash, because `NSEntityDescription.name` is actually declared as `nullable`, so if it is `nil` and we try to use it as a key in a dictionary, that will throw an exception. In case it is `nil`, we'll fall back to the prior behavior for this operation.

### CODE REVIEW

File: Sources/Sentry/SentryCoreDataTracker.m [hunk 1]
```diff
@@ -146,8 +146,9 @@ - (NSString *)descriptionForOperations:
 {
     NSMutableDictionary<NSString *, NSNumber *> *result = [NSMutableDictionary new];
 
-    for (NSManagedObject *item in entities) {
-        NSString *cl = item.entity.name;
+    for (id item in entities) {
+        // entity.name is nullable! fall back to old behavior
```
Comment (armcknight) [inline Sources/Sentry/SentryCoreDataTracker.m:150]:
```suggestion
```

**Comment by armcknight** [comment]
tests should pass after merging https://github.com/getsentry/sentry-cocoa/pull/2867

**Review by brustolin — COMMENTED** [review]
Good catch. 
I never saw entity name being nil, but since is nullable I guess is possible. 

Looks good, I think we need a test with nil entity name in order to finish this.

**Comment by armcknight** [comment]
> Good catch. I never saw entity name being nil, but since is nullable I guess is possible.
> 
> Looks good, I think we need a test with nil entity name in order to finish this.

So when I try to use the current core data tracker tests to write one like that, it throws the following exception:

> /Users/andrewmcknight/Code/organization/getsentry/repos/public/sentry-cocoa/Tests/SentryTests/Integrations/Performance/CoreData/TestCoreDataStack.swift:10: error: -[SentryTests.SentryCoreDataTrackerTests testEntityWithNilName] : Entity name must not be nil. (NSInvalidArgumentException)

**Comment by brustolin** [comment]
> So when I try to use the current core data tracker tests to write one like that, it throws the following exception

Ow, now I remember seen something like this.
Even if non nullability is enforced at run time, I don't mind keeping the changing in this PR. I guess users can do some crazy thing that would make a nil name.

File: Sources/Sentry/SentryCoreDataTracker.m [hunk 2]
```diff
@@ -146,8 +146,9 @@ - (NSString *)descriptionForOperations:
...
+        NSString *cl = ((NSManagedObject *)item).entity.name ?: NSStringFromClass([item class]);
```
Comment (brustolin) [inline Sources/Sentry/SentryCoreDataTracker.m:151]: h: I think we need to use the SwiftDescriptor for a proper swift class name demangling.

```suggestion
        NSString *cl = ((NSManagedObject *)item).entity.name ?: [SwiftDescriptor getObjectClassName:item]);
```

File: Sources/Sentry/SentryCoreDataTracker.m [hunk 3]
```diff
@@ -146,8 +147,8 @@ - (NSString *)descriptionForOperations:
 {
     NSMutableDictionary<NSString *, NSNumber *> *result = [NSMutableDictionary new];
 
-    for (NSManagedObject *item in entities) {
-        NSString *cl = item.entity.name;
+    for (id item in entities) {
```
[the remark this answers is not in the document]
Comment (brustolin) [reply Sources/Sentry/SentryCoreDataTracker.m:150]: https://github.com/getsentry/sentry-cocoa/pull/2865#issuecomment-1500012387

[the remark this answers is not in the document]
Comment (philipphofmann) [reply Sources/Sentry/SentryCoreDataTracker.m:150]: Sorry, I missed that, thanks for pointing it out @brustolin 😃

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**